### PR TITLE
Adding filtering node logic issue:321

### PR DESCRIPTION
**Pull request by Vikas-kum**
Issue: https://github.com/awslabs/sagemaker-debugger/issues/321

### Description of changes:
Adding filtering logic to dest_names and make sure that we always ask for subgraph which is present in graph def 

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smdebug/tensorflow/collection.py [hunk 1]
```diff
@@ -73,8 +73,10 @@ def add_for_mode(self, arg, mode=None):
          Adds tensors to the collection from a given Operation, Tensor, Variable or MirroredVariable
          :param arg: the argument to add to collection
          """
+        get_logger().debug(f"type:{type(arg)} arg:{arg} self:{id(self)}")
```
Comment (rahul003) [inline smdebug/tensorflow/collection.py:76]: Did you intend to remove before PR? If not, you might want to add some description before.

File: smdebug/tensorflow/collection.py [hunk 2]
```diff
@@ -124,11 +129,18 @@ def set_tensor_ref(self, tensor, tensor_name: tf.Tensor = None):
...
         self.add_tensor_name(export_name)
 
-    def has_tensor(self, name):
-        # tf object name
+        if name != export_name:
+            get_logger().debug(
+                f"Export_name:{export_name} != name:{name} . Adding export_name:{export_name} to include in "
+                f"collection collection_name:{self.name} selfId:{id(self)} "
+            )
+        self.include("^" + export_name + "$")
```
Comment (rahul003) [inline smdebug/tensorflow/collection.py:141]: This doesn't look correct. name is supposed to be the the actual name of tensor from TF's point of view. export_name is a more meaningful name for that tensor for user's benefit. Including export name is not useful and in fact might be wrong in some situations if that refers to a different tensor.

File: smdebug/tensorflow/collection.py [hunk 3]
```diff
@@ -108,7 +110,10 @@ def get_export_names_of_tensors(self):
         return self.tensor_names
 
     def get_tensor(self, name):
-        return self._tensors[name]
+        if name in self._tensors:
+            return self._tensors[name]
+        else:
+            return None
```
Comment (rahul003) [inline smdebug/tensorflow/collection.py:116]: Could you check if returning None causes issues wherever this function is used? I don't think None is expected. Is it actually possible? Did you see it happening?

File: smdebug/tensorflow/collection.py (hunk 2 above)
Comment (Vikas-kum) [reply smdebug/tensorflow/collection.py:141]: do we use export name in case of keras or for other cases also?
keras regex match works on export_name i guess.

Comment (rahul003) [reply smdebug/tensorflow/collection.py:141]: export name was only used for Keras AFAIK. @NihalHarish

Comment (rahul003) [reply smdebug/tensorflow/collection.py:141]: Why did you need this though?

Comment (Vikas-kum) [reply smdebug/tensorflow/collection.py:141]: for Tf estimator , graph is different for train and eval phase. Loss tensors are getting collected from loss collecection. 
During eval phase loss collection is empty. So we don't save any tensor. 

This change ensures that if we are saving some tensor, we add the name of the tensor being saved as regex, so that we always save these tensors if name is part of graph.

Comment (Vikas-kum) [reply smdebug/tensorflow/collection.py:141]: So, question is if export_name is there, is it used for searching collection regex? 
In other words - 
in what cases tensor_name won't be there passed to this function and 
what(name or export name?) is being used for searching collection regex in these cases ?
